File: Data Structures & Algorithms/redundant-connection/submission-2.py
class Solution:
    def findRedundantConnection(self, edges: List[List[int]]) -> List[int]:
        def dfs(node):
            if node == a:
                return True
            rec_nodes[node] = True
            
            for edge in graph[node]:
                if not rec_nodes[edge]:
                    if dfs(edge):
                        return True

            return False

        graph = {i: [] for i in range(len(edges)+1)}
        res = [0,0]
        for a,b in edges:
            rec_nodes = [False] * (len(edges)+1)
            graph[a].append(b)
            if dfs(b):
                res[0], res[1] = a,b
            graph[b].append(a)

        return res

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
public class Solution {
    private void DFS(TreeNode root, List<List<int>> list, int level){
        if (root == null) return;
        if (level >= list.Count){
            list.Add(new List<int>());
        }

        list[level].Add(root.val);
        DFS(root.left, list, level+1);
        DFS(root.right, list, level+1);
    }

    public List<List<int>> LevelOrder(TreeNode root) {
        var res = new List<List<int>>();
        DFS(root, res, 0);
        return res;
    }
}
